### ISSUE TITLE
Applying interface with type parameter on module without type parameter

### ISSUE THREAD

**Issue by rand00** [ocaml/ocaml#11610]
Compiling the following with `ocamlbuild bug.byte` prints:
```
+ ocamlfind ocamlc -c -o bug.cmo bug.ml
Fatal error: exception Invalid_argument("List.iter2")
Command exited with code 2.
Compilation unsuccessful after building 2 targets (0 cached) in 00:00:00.
```

```ocaml
module type S = sig

  type 'a error = unit
  val get : unit -> _ error

end

module M : S = struct

  (* type 'a error = unit (\*< correct type*\)*)
  type error = unit
  let get () = failwith "not implemented"

end
```
.. and there is no error reported by Merlin in the code either

**Comment by Octachron** [comment]
This a 4.13 only bug (see #10836) that has been fixed in 4.14 (see #10952).

**Comment by Octachron** [comment]
I forgot on the spur of the moment: thanks for the report!
